### ISSUE TITLE
Support `Literal` with `NewType`

### ISSUE THREAD

**Issue by yaxum62** [microsoft/pyright#4149]
Given a NewType from primitive type, e.g.

```python
UserId = NewType("UserId", int)
```

Current pyright reports error on attempts of using such NewType as Literal:

```python
INVALID_USER_ID: Literal[UserId(0)] = UserId(0)
                       # ^^^^^^^^^Type arguments for "Literal" must be None, a literal value (int, bool, str, or bytes), or an enum valuePylance
```

The document of `NewType` mentioned that `UserId` will always return argument as-is, thus it should be safe to assume that its return value will meet whatever requirements on `Literal` argument.

**Comment by erictraut** [comment]
`Literal` is a special form introduced in [PEP 586](https://peps.python.org/pep-0586/). The PEP specifically calls out the types that can be used: int, byte, str, bool, None, Enum values. It does not allow subtypes of any of these, including subtypes created with `NewType`.

**Comment by yaxum62** [comment]
Thanks for your review!

PEP 586 listed those primitive types as legal case, but it didn't mention that all other cases are illegal. Instead it specifically listed few cases as illegal and proposed case is not there. 

Seem proposed case is more like a case of `type checkers may impose restrictions`, as mentioned in `typing.Literal` document. I hope such restrictions can be lifted with this feature request.
